File: app/src/main/java/com/tiktokaitool/SettingsActivity.java
package com.tiktokaitool;
import android.os.Bundle;
import android.widget.*;
import androidx.appcompat.app.AppCompatActivity;

public class SettingsActivity extends AppCompatActivity {
    private EditText etKey, etPrompt;
    private Spinner spinModel;
    private Switch swAutoOpen;
    private PrefsManager prefs;
    private final String[] models = {"claude-opus-4-5","claude-sonnet-4-5","claude-haiku-4-5-20251001"};
    private final String[] labels = {"Claude Opus 4.5 (Best)","Claude Sonnet 4.5 (Fast)","Claude Haiku 4.5 (Fastest)"};

    @Override protected void onCreate(Bundle b) {
        super.onCreate(b);
        setContentView(R.layout.activity_settings);
        prefs = new PrefsManager(this);
        etKey    = findViewById(R.id.et_api_key);
        etPrompt = findViewById(R.id.et_prompt);
        spinModel= findViewById(R.id.spin_model);
        swAutoOpen=findViewById(R.id.sw_auto_open);

        etKey.setText(prefs.getApiKey());
        etPrompt.setText(prefs.getPrompt());
        swAutoOpen.setChecked(prefs.isAutoOpen());

        ArrayAdapter<String> adapter = new ArrayAdapter<>(this,
            android.R.layout.simple_spinner_item, labels);
        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
        spinModel.setAdapter(adapter);
        String cur = prefs.getModel();
        for (int i=0;i<models.length;i++) if(models[i].equals(cur)){spinModel.setSelection(i);break;}

        findViewById(R.id.btn_save).setOnClickListener(v -> save());
        findViewById(R.id.btn_back).setOnClickListener(v -> finish());
        findViewById(R.id.btn_reset_prompt).setOnClickListener(v -> {
            prefs.setPrompt(""); etPrompt.setText(prefs.getPrompt());
        });
    }

    private void save() {
        String key = etKey.getText().toString().trim();
        if (key.isEmpty()) { Toast.makeText(this,"API key required",Toast.LENGTH_SHORT).show(); return; }
        prefs.setApiKey(key);
        prefs.setPrompt(etPrompt.getText().toString().trim());
        prefs.setModel(models[spinModel.getSelectedItemPosition()]);
        prefs.setAutoOpen(swAutoOpen.isChecked());
        Toast.makeText(this,"✓ Saved!",Toast.LENGTH_SHORT).show();
        finish();
    }
}

File: app/src/main/java/com/tiktokaitool/ui/SettingsActivity.kt
package com.tiktokaitool.ui

import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.tiktokaitool.api.ClaudeAnalyzer
import com.tiktokaitool.databinding.ActivitySettingsBinding
import com.tiktokaitool.utils.PrefsManager

class SettingsActivity : AppCompatActivity() {

    private lateinit var binding: ActivitySettingsBinding
    private lateinit var prefs: PrefsManager

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivitySettingsBinding.inflate(layoutInflater)
        setContentView(binding.root)

        prefs = PrefsManager(this)

        binding.toolbar.setNavigationOnClickListener { finish() }

        // Load current values
        binding.etApiKey.setText(prefs.getApiKey())
        binding.etCustomPrompt.setText(prefs.getAnalysisPrompt().ifBlank { ClaudeAnalyzer.DEFAULT_PROMPT })
        binding.switchOverlay.isChecked = prefs.isOverlayEnabled()
        binding.switchAutoOverlay.isChecked = prefs.isAutoOverlayOnTikTok()

        // Focus on API key if coming from warning
        if (intent.getBooleanExtra("focus_api_key", false)) {
            binding.etApiKey.requestFocus()
        }

        binding.btnSaveSettings.setOnClickListener {
            val apiKey = binding.etApiKey.text.toString().trim()
            val prompt = binding.etCustomPrompt.text.toString().trim()

            if (apiKey.isBlank()) {
                Toast.makeText(this, "API key cannot be empty", Toast.LENGTH_SHORT).show()
                return@setOnClickListener
            }

            prefs.saveApiKey(apiKey)
            prefs.saveAnalysisPrompt(prompt)
            prefs.setOverlayEnabled(binding.switchOverlay.isChecked)
            prefs.setAutoOverlayOnTikTok(binding.switchAutoOverlay.isChecked)

            Toast.makeText(this, "✅ Settings saved!", Toast.LENGTH_SHORT).show()
            finish()
        }

        binding.btnResetPrompt.setOnClickListener {
            binding.etCustomPrompt.setText(ClaudeAnalyzer.DEFAULT_PROMPT)
            Toast.makeText(this, "Prompt reset to default", Toast.LENGTH_SHORT).show()
        }

        binding.tvGetApiKey.setOnClickListener {
            val intent = android.content.Intent(
                android.content.Intent.ACTION_VIEW,
                android.net.Uri.parse("https://console.anthropic.com/")
            )
            startActivity(intent)
        }
    }
}
